keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -108,11 +105,15 @@ class ArtifactListener(
                 artifactVersion = artifact.version,
                 type = BUILD,
                 id = "build-${artifact.version}",
-                status = NOT_STARTED,
+                // the build has already started, and is maybe complete.
+                // We use running to convey that to users, and allow the [BuildLifecycleMonitor] to immediately
+                // update the status
```

Comment:
Can you explain a bit more on how the build monitor will update the status immediately if the build completed?

Reply:
The build monitor does a status check immediately after the event published (that's just how it works because the event has never been monitored). When it does it'll figure out what the actual status is and then publish another lifecycle event with the correct status and user link.

Reply:
I see the logic there, but as we discussed in your previous PR, the status can be terminal here. Isn't it worth passing here the complete build metadata object, and then determinate if to check the status again?

Reply:
Alright, I made that change!